From 751eeb889b13ba0c7976f93bc9195b77e2c7f38a Mon Sep 17 00:00:00 2001
From: Archith John <archith@ti.com>
Date: Fri, 11 Mar 2011 16:47:38 +0530
Subject: [PATCH] TI814x: Enable interleaving and set IVA clock speed for normal device

Enable DMM inter-leaving and set IVA clock speed to normal device value.

Signed-off-by: Sriramakrishnan <srk@ti.com>
---
 arch/arm/include/asm/arch-ti81xx/clocks_ti814x.h |    4 ++--
 board/ti/ti8148/evm.c                            |   18 +++++++++---------
 2 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/arch/arm/include/asm/arch-ti81xx/clocks_ti814x.h b/arch/arm/include/asm/arch-ti81xx/clocks_ti814x.h
index 017403b..46982ff 100644
--- a/arch/arm/include/asm/arch-ti81xx/clocks_ti814x.h
+++ b/arch/arm/include/asm/arch-ti81xx/clocks_ti814x.h
@@ -48,12 +48,12 @@
 #define DSP_CLKCTRL	0x801
 
 #define DSS_N		19
-#define DSS_M		666
+#define DSS_M		800
 #define DSS_M2		4
 #define DSS_CLKCTRL	0x801
 
 #define IVA_N		19
-#define IVA_M		850
+#define IVA_M		640
 #define IVA_M2		2
 #define IVA_CLKCTRL	0x801
 
diff --git a/board/ti/ti8148/evm.c b/board/ti/ti8148/evm.c
index cc71c9a..a392348 100644
--- a/board/ti/ti8148/evm.c
+++ b/board/ti/ti8148/evm.c
@@ -193,15 +193,15 @@ static void config_ti814x_ddr(void)
 	while( (__raw_readl(VTP1_CTRL_REG) & 0x00000020) != 0x20);
 
 	/*Program the DMM to Access EMIF0 and EMIF1*/
-	__raw_writel(0x80600100, DMM_LISA_MAP__0);
-	__raw_writel(0x80600100, DMM_LISA_MAP__1);
-	__raw_writel(0xC0600200, DMM_LISA_MAP__2);
-	__raw_writel(0xC0600200, DMM_LISA_MAP__3);
+	__raw_writel(0x80440300, DMM_LISA_MAP__0);
+	__raw_writel(0x80440300, DMM_LISA_MAP__1);
+	__raw_writel(0xC0440308, DMM_LISA_MAP__2);
+	__raw_writel(0xC0440308, DMM_LISA_MAP__3);
 
-	while(__raw_readl(DMM_LISA_MAP__0)!=0x80600100);
-	while(__raw_readl(DMM_LISA_MAP__1)!=0x80600100);
-	while(__raw_readl(DMM_LISA_MAP__2)!=0xC0600200);
-	while(__raw_readl(DMM_LISA_MAP__3)!=0xC0600200);
+	while(__raw_readl(DMM_LISA_MAP__0)!=0x80440300);
+	while(__raw_readl(DMM_LISA_MAP__1)!=0x80440300);
+	while(__raw_readl(DMM_LISA_MAP__2)!=0xC0440308);
+	while(__raw_readl(DMM_LISA_MAP__3)!=0xC0440308);
 
 	__raw_writel(0x80000000, DMM_PAT_BASE_ADDR);
 
@@ -499,7 +499,7 @@ void prcm_init(u32 in_ddr)
 	l3_pll_config();
 	ddr_pll_config();
 	dsp_pll_config();
-	dss_pll_config();
+//	dss_pll_config();
 	iva_pll_config();
 	iss_pll_config();
 
-- 
1.6.2.4

